chesster.js

```diff
@@ -1312,224 +516,244 @@ function scheduling_reply_scheduled(bot, message, results, white, black) {
     var user = "<@"+message.user+">";
     bot.reply(message, ":x: " + user + " I don't recognize one of the players you mentioned.");
 }
 
 // Scheduling will occur on any message
-controller.on('ambient', function(bot, message) {
-    bot_exception_handler(bot, message, function(){
-        var channel = channels.byId[message.channel];
-        if (!channel) {
-            return;
-        }
-        var scheduling_options = config.scheduling[channel.name];
-        if (!scheduling_options) {
-            return;
-        } 
-
-        var is_potential_schedule = false;
-        var references_slack_users = false;
-        var has_pairing = false;
-
-        var results = {
-            white: '',
-            black: ''
-        };
-
-        // Step 1. See if we can parse the dates
-        try {
-            results = spreadsheets.parse_scheduling(message.text, scheduling_options);
-            is_potential_schedule = true;
-        } catch (e) {
-            if (e instanceof (spreadsheets.ScheduleParsingError)) {
-            } else {
-                throw e; // let others bubble up
-            }
-        }
+chesster.on({
+    event: 'ambient',
+    middleware: [slack.requiresLeague]
+},
+function(bot, message) {
+    var channel = channels.byId[message.channel];
+    if (!channel) {
+        return;
+    }
+    var schedulingOptions = message.league.options.scheduling;
+    if (!schedulingOptions) {
+        console.error("{} league doesn't have scheduling options!?".format(message.league.options.name));
+        return;
+    } 
+    var spreadsheetOptions = message.league.options.spreadsheet;
+    if (!spreadsheetOptions) {
+        console.error("{} league doesn't have spreadsheet options!?".format(message.league.options.name));
+        return;
+    } 
+    if (channel.name != schedulingOptions.channel) {
+        return;
+    }
 
-        // Unless they included a date we can parse, ignore this message.
-        if (!is_potential_schedule) {
-            return;
-        }
+    var isPotentialSchedule = false;
+    var referencesSlackUsers = false;
+    var hasPairing = false;
+
+    var results = {
+        white: '',
+        black: ''
+    };
 
-        // Step 2. See if we have valid named players
-        var white = users.getByNameOrID(results.white);
-        var black = users.getByNameOrID(results.black);
-        if (white && black) {
-            results.white = white.name;
-            results.black = black.name;
-            references_slack_users = true;
+    // Step 1. See if we can parse the dates
+    try {
+        results = spreadsheets.parseScheduling(message.text, schedulingOptions);
+        isPotentialSchedule = true;
+    } catch (e) {
+        if (e instanceof (spreadsheets.ScheduleParsingError)) {
+        } else {
+            throw e; // let others bubble up
         }
+    }
 
+    // Unless they included a date we can parse, ignore this message.
+    if (!isPotentialSchedule) {
```

Comment:
Why dont we just return in the catch? seems this will only false if the catch is executed.